api/v1/controllers/generatorTemplates.js

```diff
@@ -49,6 +50,20 @@ module.exports = {
     doGet(req, res, next, helper);
   },
 
+  /**
+   * GET /generatorTemplates/{name}/{version}
+   *
+   * Retrieves the generatorTemplate by name and version and sends it back
+   * in the response.
+   *
+   * @param {IncomingMessage} req - The request object
+   * @param {ServerResponse} res - The response object
+   * @param {Function} next - The next middleware function in the stack
+   */
+  getGeneratorTemplateByNameAndVersion(req, res, next) {
+    doFindOne(req, res, next, helper);
```

Comment:
Is it necessary to set up a whole new helper for this? We're already enforcing that a name/version pair is unique, so we can be sure we would only get one result if we use doFind.

Reply:
doFind returns an array and the expectation is the object.

Reply:
Hmm. Makes sense, that would be weird.
Not ideal, but this is probably the best we can do short of completely refactoring the helpers.

We could at least get rid of any code in the new helper that not's necessary for this case, like the caching.

Reply:
yeap! We have to refactor those helpers and revisit some steps.

Thx for pointed out the cache, if you see something else let me know. I don't have to much context to deliberately remove them.